src/client/app/utils/api/UploadCSVApi.ts

```diff
@@ -4,53 +4,75 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import ApiBackend from './ApiBackend';
+import { Dispatch } from '@reduxjs/toolkit';
+import { baseApi } from '../../redux/api/baseApi';
 import {
-	ReadingsCSVUploadPreferencesItem, MetersCSVUploadPreferencesItem, BooleanTypes, CSVUploadPreferencesForm, ReadingsCSVUploadPreferencesForm
+	CSVUploadPreferencesForm,
+	MetersCSVUploadPreferencesItem,
+	ReadingsCSVUploadPreferencesForm,
+	ReadingsCSVUploadPreferencesItem
 } from '../../types/csvUploadForm';
+import ApiBackend from './ApiBackend';
+import { TrueFalseType } from '../../types/items';
+
+interface ApiResponse {
+	success: boolean,
+	message: string
+}
 
-export default class UploadCSVApi {
-	private readonly backend: ApiBackend;
+export const submitReadings = async (uploadPreferences: ReadingsCSVUploadPreferencesItem, readingsFile: File,
+	dispatch: Dispatch): Promise<ApiResponse> => {
+	const backend = new ApiBackend();
+	const formData = new FormData();
+	// The Boolean values in state must be converted to the submitted values of yes and no.
+	const uploadPreferencesForm: ReadingsCSVUploadPreferencesForm = {
+		...uploadPreferences,
+		gzip: uploadPreferences.gzip ? TrueFalseType.true : TrueFalseType.false,
```

Comment:
When I try to upload a file I get an enum error. I did not try to track down where it is coming from. Whatever changes are made, we also need to be sure that the curl commands work.

I also notice that BooleanTypes was removed from the client but is in many places in the server code. I supect this related.

Reply:
I am no longer able to reproduce this error. Let me know if you are still getting it.

Reply:
This is the error I see:

![image](https://github.com/user-attachments/assets/09bfba56-83ec-492c-b79f-079e4a4ab0fa)

I still see the error and I think I have debugged it. You converted the client CSVUploads to be true/false. That maps true/false to yes/no which is good for the display. The issue is that the server still expected yes/no. See src/server/services/csvPipeline/validateCsvUploadParams.js for BooleanTypesJS which is used as the allowed values for cumulative, etc. I believe I did this a while back to show yes/no to the user but it was not the best way to do it. Now the enum loop shows that value but sends true/false. This is better since internally the value is a boolean. I verified this in two ways:

1. I put a console.log in src/server/services/csvPipeline/validateCsvUploadParams.js at the start of validateReadingsCsvUploadParams to show the values sent were true/false.
2. I changed `cumulative: new EnumParam('cumulative', [BooleanTypesJS.true, BooleanTypesJS.false]),` to `cumulative: Boolean,` and the error went away (there are other errors for the other params that have a similar issue.

Thus, I think this type of change will resolve the issue and improve the code from what it was before your changes. Please change for all the types needed, if you agree, and let me know if you think otherwise.

As long as I was deep in the code, I also wanted to double check about what happens if a direct (e.g., curl) request is made. In src/server/services/csvPipeline/uploadReadings.js on line 86 is checks if cumulative is not provided and then uses the meter value (assuming it exists). So, you can still get the meter value by not providing the value. This is what is already documented in https://openenergydashboard.org/helpV1_0_0/adminReadingsImport/ so no changes are needed. I'll add that what you said before about checking it was correct (I'm just paranoid :-).

Reply:
I'm unresolving since I still see this. I was doing final testing when I saw this.

Reply:
Got it, I will work on fixing this.

Reply:
I have fixed this issue as far as I can tell. I created a utility function called normalizeBoolean which took any 'yes' 'no' values and converted to boolean.